doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -568,67 +493,50 @@ Compiled files
+     in the module available by their short names
+   * :n:`Export` - additionally does an :cmd:`Export` on the loaded module, making components defined
+     in the module available by their short names *and* marking them to be exported by the current
+     module
 
-      This loads and declares the module :n:`@qualid`
-      and its dependencies then imports the contents of :n:`@qualid` as described
-      for :cmd:`Import`. It does not import the modules that
-      :n:`@qualid` depends on unless these modules were themselves required in module
-      :n:`@qualid`
-      using :cmd:`Require Export`, or recursively required
-      through a series of :cmd:`Require Export`.  If the module required has
-      already been loaded, :cmd:`Require Import` :n:`@qualid` simply imports it, as
-      :cmd:`Import` :n:`@qualid` would.
+   If the required module has already been loaded, :n:`Import` and :n:`Export` make the command
+   equivalent to :cmd:`Import` or :cmd:`Export`.
 
-   .. cmdv:: Require Export @qualid
-      :name: Require Export
-
-      This command acts as :cmd:`Require Import` :n:`@qualid`,
-      but if a further module, say `A`, contains a command :cmd:`Require Export` `B`,
-      then the command :cmd:`Require Import` `A` also imports the module `B.`
+   The mapping between
+   physical directories and logical names at the time of requiring the
+   file must be consistent with the mapping used to compile the file. If
+   several files match, one of them is picked in an unspecified fashion.
 
-   .. cmdv:: Require {| Import | Export } {+ @qualid }
+   .. todo common user error on dirpaths see https://github.com/coq/coq/pull/11961#discussion_r402852390
 
-      This loads the
-      modules named by the :token:`qualid` sequence and their recursive
-      dependencies. If
-      ``Import`` or ``Export`` is given, it also imports these modules and
-      all the recursive dependencies that were marked or transitively marked
-      as ``Export``.
+.. cmd:: From @qualid Require {? {| Import | Export } } {+ @qualid }
+   :name: From ... Require ...
 
-   .. cmdv:: From @dirpath Require @qualid
-      :name: From ... Require ...
+   .. todo: wanted to change @qualid to @dirpath
+      but even without that, I don't get the description with ":n:`@dirpath.@dirpath’.@qualid`"
 
-      This command acts as :cmd:`Require`, but picks
-      any library whose absolute name is of the form :n:`@dirpath.@dirpath’.@qualid`
-      for some :n:`@dirpath’`. This is useful to ensure that the :token:`qualid` library
-      comes from a given package by making explicit its absolute root.
+   Works like :cmd:`Require`, but picks
+   any library whose absolute name is of the form :n:`@dirpath.@dirpath’.@qualid`
+   for some :n:`@dirpath’`. This is useful to ensure that the :token:`qualid` library
+   comes from a given package by making explicit its absolute root.
```

Comment:
From instance, `From Ltac2 Require List` is equivalent to `Require Ltac2.List` while `From Coq Require List` is equivalent to `Require Coq.Lists.List`.

Reply:
"any library whose absolute name is":
- "any library" - it only loads one library for each `{+ @dirpath }`, right?
- is "absolute" correct?  How about "fully qualified name"?
- as with Require, you're in trouble if it's still not unique

Reply:
>"any library" - it only loads one library for each `{+ @dirpath }`, right?

Yes.

>is "absolute" correct? How about "fully qualified name"?

Maybe replace "absolute" with "complete". "Fully qualified" would also work but could bring confusion with qualified identifiers.

>as with Require, you're in trouble if it's still not unique

Yes, but it seriously reduces the chances of collision. Collisions could only occur if:
- the author of the library / current project (if we are requiring from the current project) has named several modules the same (strongly discouraged, and it would be great if we could even raise an error when that's the case---maybe worth an issue about this).
- several libraries have the same base name (this is the job of the package index maintainers to ensure that this is not the case, see also coq/opam-coq-archive#1208).